provider/offline.go

```diff
@@ -1571,3 +1572,44 @@ func genericIterateChunk(mat Materialization, rowsPerChunk int64, idx int) (Feat
 	}
 	return mat.IterateSegment(start, end)
 }
+
+func (def *TrainingSetDef) ToBuilderParams(logger logging.Logger, sanitizeTableNameFn func(pl.Location) (string, error)) (tsq.BuilderParams, error) {
+	lblTableName, err := sanitizeTableNameFn(def.LabelSourceMapping.Location)
```

Comment:
This whole method is extremely hard to follow, try using better names, breaking it up, etc.